plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/helpers.ts

```diff
@@ -43,18 +43,93 @@ export interface Results {
   results: Result[];
 }
 
+export type ConfluenceConfig = {
+  baseUrl: string;
+  auth: string;
+  token?: string;
+  email?: string;
+  username?: string;
+  password?: string;
+};
+
+export const getConfluenceConfig = (config: Config) => {
+  const confluenceConfig: ConfluenceConfig = {
+    baseUrl: config.getString('confluence.baseUrl'),
+    auth: config.getOptionalString('confluence.auth') ?? 'bearer',
+    token: config.getOptionalString('confluence.token'),
+    email: config.getOptionalString('confluence.email'),
+    username: config.getOptionalString('confluence.username'),
+    password: config.getOptionalString('confluence.password'),
+  };
+
+  if (
+    (confluenceConfig.auth === 'basic' || confluenceConfig.auth === 'bearer') &&
+    !confluenceConfig.token
+  ) {
+    throw new Error(
+      `No token provided for the configured '${confluenceConfig.auth}' auth method`,
+    );
+  }
+
+  if (confluenceConfig.auth === 'basic' && !confluenceConfig.email) {
+    throw new Error(
+      `No email provided for the configured '${confluenceConfig.auth}' auth method`,
+    );
+  }
+
+  if (
+    confluenceConfig.auth === 'userpass' &&
+    (!confluenceConfig.username || !confluenceConfig.password)
+  ) {
+    throw new Error(
+      `No username/password provided for the configured '${confluenceConfig.auth}' auth method`,
+    );
+  }
+
+  return confluenceConfig;
+};
+
+export const getAuthorizationHeaderValue = (config: ConfluenceConfig) => {
+  let authHeaderValue: string = '';
```

Comment:
i'd just delete this line and return in each case instead, no need for breaks then too :)

Reply:
Sounds good, done 👍